ThinDownloadManager/src/main/java/com/thin/downloadmanager/DownloadRequestQueue.java

```diff
@@ -230,9 +233,13 @@ void pauseAll() {
 	private void checkResumableDownloadEnabled(int downloadId) {
 		synchronized (mCurrentRequests) {
 			for (DownloadRequest request : mCurrentRequests) {
-					if (downloadId == -1 ||
-							(request.getDownloadId() == downloadId && !request.isResumable())) {
-						throw new IllegalStateException("You cannot pause the download, unless your request enabled Resume feature in DownloadRequest.");
+				if (downloadId == -1 && !request.isResumable()) {
+					Log.e("ThinDownloadManager",
+							String.format(Locale.getDefault(), "This request has not enabled resume feature hence request will be cancelled. Request Id: %d", request.getDownloadId()));
+				} else if ((request.getDownloadId() == downloadId && !request.isResumable())) {
+					throw new IllegalStateException("You cannot pause the download, unless you have enabled Resume feature in DownloadRequest.");
```

Comment:
@smanikandan14 
**Usage of the line:** This is exception will help user to identify the issue reason exactly, what was gone wrong!